torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -1,72 +1,101 @@
-from typing import Tuple, List, Optional
+from typing import TypeVar, Any, cast
 
+import PIL.Image
 import torch
-from torchvision.transforms import (  # noqa: F401
-    functional as _F,
-    InterpolationMode,
+from torchvision.prototype import features
+from torchvision.prototype.transforms import kernels as K
+from torchvision.transforms import functional as _F
+
+from ._utils import dispatch
+
+T = TypeVar("T", bound=features.Feature)
+
+
+@dispatch(
+    {
+        torch.Tensor: _F.hflip,
+        PIL.Image.Image: _F.hflip,
+        features.Image: K.horizontal_flip_image,
+        features.BoundingBox: None,
+    },
 )
+def horizontal_flip(input: T, *args: Any, **kwargs: Any) -> T:
+    """ADDME"""
+    if isinstance(input, features.BoundingBox):
+        output = K.horizontal_flip_bounding_box(input, format=input.format, image_size=input.image_size)
+        return cast(T, features.BoundingBox.new_like(input, output))
+
+    raise RuntimeError(
```

Comment:
I would just throw a simple RuntimeError() here without text (to avoid needing to maintain all these messages. These should be unreachable code if the dispatcher works right.